src/main/java/com/aws/greengrass/device/iot/IotAuthClient.java

```diff
@@ -19,22 +19,36 @@
 import software.amazon.awssdk.services.greengrassv2data.model.VerifyClientDeviceIdentityResponse;
 import software.amazon.awssdk.services.greengrassv2data.model.VerifyClientDeviceIoTCertificateAssociationRequest;
 
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.time.Duration;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
 import javax.inject.Inject;
 
 public interface IotAuthClient {
     Optional<String> getActiveCertificateId(String certificatePem);
 
     boolean isThingAttachedToCertificate(Thing thing, Certificate certificate);
 
+    void clearLocalAuthCache();
+
     class Default implements IotAuthClient {
         private static final Logger logger = LogManager.getLogger(Default.class);
         private static final RetryUtils.RetryConfig SERVICE_EXCEPTION_RETRY_CONFIG =
                 RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofMillis(100)).maxAttempt(3)
                         .retryableExceptions(Arrays.asList(ThrottlingException.class, InternalServerException.class))
                         .build();
+        // TODO: clear local auth-cache as necessary
```

Comment:
Memory considerations aside, I can't think of a reason we'd ever need to clear PEM -> Cert ID mappings. IDs are immutable and the chances of two distinct certs hashing to the same ID are infinitesimally small.

Reply:
We consider a certificate "active" when IoT Core returns corresponding Certificate ID for its pem. When user manually _deactivates_ a certificate from the console, does IoT Core removes its CertID from its registry? Should this cached CertPem <-> CertId mapping be still valid then?

Reply:
CertID is just a hash computed from the certificate itself. Metadata about the certificate can change without the ID becoming invalid so keeping it is harmless.

It comes back to the point of this cache. It is NOT to prevent cloud calls if a device disconnects and reconnects. We are doing that later.

Reply:
PR updated, thanks.